app/Http/Controllers/Api/AssetsController.php

```diff
@@ -994,34 +1013,18 @@ public function checkin(Request $request, $asset_id): JsonResponse
         if ($request->filled('status_id')) {
             $asset->status_id = $request->input('status_id');
         }
-
-        $checkin_at = $request->filled('checkin_at') ? $request->input('checkin_at') . ' ' . date('H:i:s') : date('Y-m-d H:i:s');
-        $originalValues = $asset->getRawOriginal();
-
+        
         if (($request->filled('checkin_at')) && ($request->get('checkin_at') != date('Y-m-d'))) {
-            $originalValues['action_date'] = $checkin_at;
+            \Log::error("Setting checkin time of ".$request->get('checkin_at')." because that's not today which is: ".date('Y-m-d'));
+            $asset->setLogDate(new Carbon($request->get('checkin_at')));
```

Comment:
If this only affects action_date, can we rename it to be more specific, like `setLogActionDate`?